### PR TITLE
TASK-66374: Introduce crowdin github action workflow

### PR DESCRIPTION
**Pull request by bourasom**
This PR aims to introduce a github action workflow sync with crowdin:

- Uploads are scheduled up to push on develop, develop-exo or develop-meed branch. They consider the fact that .properties files were concerned by changes
- Once a day, the translations are downloaded from crowdin and a merge is applied on the concerned branches.
- This PR assumes that:
   -  develop is synchronised with the main crowdin project
   -  develop-meed is synchronised with develop-meed crowdin project's branch
   -  develop-exo is synchronised with develop-exo crowdin project's branch
- For the WF purposes, a branches integration/6.5.x-translation, integration/6.5.x-meed-translation and integration/6.5.x-exo-translation will be created from respectively develop, develop-meed and develop-exo

### CODE REVIEW

File: .github/workflows/download-crowdin.yml [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+          token: ${{ secrets.CROWDIN_GITHUB_TOKEN }}
+      - name: reset translation branch to develop's
+        run: |
+          git fetch origin develop
+          git config --global user.email "exo-swf@exoplatform.com"
+          git config --global user.name "eXo Software Factory"
+          git reset origin/develop
+          git add .
+          git stash
+          git push origin -f
```
Comment (hbenali) [inline .github/workflows/download-crowdin.yml]:
```suggestion
          git push origin origin/develop:refs/heads/integration/6.5.x-translation -f
```

**Review by hbenali — COMMENTED** [review]
The integration branches could be deleted. So we can't checkout it. This proposal will ensure its creation. To apply for others also develop-meed & develop-exo if you accept it. Moreover we don't need fetch to keep the shallow clone default behavior of `actions/checkout@v3` workflow.

File: .github/workflows/download-crowdin.yml [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+jobs:
+  # develop workflow
+  reset-branch-integration-develop:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: integration/6.5.x-translation
```
Comment (hbenali) [inline .github/workflows/download-crowdin.yml]:
```suggestion
          ref: develop
```

**Comment by bourasom** [comment]
> The integration branches could be deleted. So we can't checkout it. This proposal will ensure its creation. To apply for others also develop-meed & develop-exo if you accept it. Moreover we don't need fetch to keep the shallow clone default behavior of `actions/checkout@v3` workflow.

Yes, I totally agree, thanks (y)

**Review by hbenali — DISMISSED** [review]

**Review by hbenali — DISMISSED** [review]

File: .github/workflows/upload-crowdin-main.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      crowdin_project_available: ${{ steps.is_crowdin_project_available.outputs.project_available }}
+    steps:
+    - name: Check project build in progress
+      id: is_crowdin_project_available
+      run: |
+        wait_exec=0;
+        while [ $(curl -s -H "Accept: application/json" -H "Authorization: Bearer ${{ secrets.CROWDIN_PERSONAL_TOKEN }}"\
+                                                                     https://crowdin.com/api/v2/projects/4464/translations/builds \
```
Comment (plamarque) [inline .github/workflows/upload-crowdin-main.yml]: should not be hardcoded

File: .github/workflows/download-crowdin.yml [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      crowdin_project_available: ${{ steps.is_crowdin_project_available.outputs.project_available }}
+    steps:
+      - name: Check project build in progress
+        id: is_crowdin_project_available
+        run: |
+          wait_exec=0;
+          while [ $(curl -s -H "Accept: application/json" -H "Authorization: Bearer ${{ secrets.CROWDIN_PERSONAL_TOKEN }}"\
+                                                                       https://crowdin.com/api/v2/projects/4464/translations/builds \
```
Comment (plamarque) [inline .github/workflows/download-crowdin.yml]: should not be hardcoded

File: .github/workflows/upload-crowdin-branches.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      crowdin_project_available: ${{ steps.is_crowdin_project_available.outputs.project_available }}
+    steps:
+    - name: Check project build in progress
+      id: is_crowdin_project_available
+      run: |
+        wait_exec=0;
+        while [ $(curl -s -H "Accept: application/json" -H "Authorization: Bearer ${{ secrets.CROWDIN_PERSONAL_TOKEN }}"\
+                                                                     https://crowdin.com/api/v2/projects/4464/translations/builds \
```
Comment (plamarque) [inline .github/workflows/upload-crowdin-branches.yml]: should not be hardcoded

Comment (bourasom) [reply .github/workflows/upload-crowdin-branches.yml]: fixed

File: .github/workflows/download-crowdin.yml (hunk 4 above)
Comment (bourasom) [reply .github/workflows/download-crowdin.yml]: fixed

File: .github/workflows/upload-crowdin-main.yml (hunk 3 above)
Comment (bourasom) [reply .github/workflows/upload-crowdin-main.yml]: fixed

**Comment by bourasom** [comment]
@rdenarie please let me know when you had a view on the PR like agreed
@plamarque the dests parts like this one https://github.com/Meeds-io/gamification/pull/1218/files#diff-256aa18d225b85dd7da40aba5091225c2b6c93e8e40f8955fc92be6c5a674afcR31 need to be changed according to the target that you need in the newly crowdin meed project. Please share with me the structure that you want to apply

File: crowdin.yml [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    "translation" : "%original_path%/%file_name%!_%locale_with_underscore%.%file_extension%",
+    "translation_replace" : {
+      "_en!":"",
+      "de_DE": "de",
+      "en_US": "en",
+      "es_ES": "es_ES",
+      "fr_FR": "fr",
+      "it_IT": "it",
+      "pt_BR": "pt_BR",
```
Comment (rdenarie) [inline crowdin.yml]: In the translation.zip file, we have much more languages than theses one.
I think we must put all in this list to make it work.
```suggestion
      "ar_SA":"ar",
      "ar_OM":"aro",
      "az_AZ":"az",
      ....
      "de_DE": "de",
      "en_US": "en",
      "es_ES": "es_ES",
      "fr_FR": "fr",
      "it_IT": "it",
      "pt_BR": "pt_BR",
```
I put just the main ones. 
As this is only for download action, it is possible to test this specific action : 
launch it, but not merge. So we can confirm that in the created merge branch, all languages are ok, and fix if necessary
Should we merge before test ? or can we test before merge ?

Comment (rdenarie) [reply crowdin.yml]: I use this page to know the main regional parameter for each languages :+1: https://help.sap.com/docs/SAP_BUSINESSOBJECTS_BUSINESS_INTELLIGENCE_PLATFORM/09382741061c40a989fae01e61d54202/46758c5e6e041014910aba7db0e91070.html

Comment (bourasom) [reply crowdin.yml]: the merge should be done in order to apply the github actions. I may comment the merge part in the begining

Comment (rdenarie) [reply crowdin.yml]: Ok, So my proposition is to do that : 
- comment merge part
- Check download part to answer this question : does all languages are ok with this configuration, or should we add all languages codes in crowdin.yml ?
- If ok, uncomment merge part and deactivate jenkins sybc/merge for gamification

Comment (bourasom) [reply crowdin.yml]: done

**Review by hbenali — APPROVED** [review]

**Pull request merged**